### PR TITLE
Bug 2085997: Report late requests verification: Fix for Hypershift

### PR DESCRIPTION
**Pull request by alvaroaleman**
The late requests validation is based on interpreting the KAS audit
logs. The audit logs are fetched through `oc must-gather`. This works
fine for self-hosted OCP, but with an external controlplane the KAS runs
as pod in a distinct management cluster and thus it is not possible to
get its audit logs from the cluster under test.

This change fixes the test for Hypershift by introduction dedicated
audit log fetching codepaths for selfhosted/hypershift.

/assign @p0lyn0mial 

The diff gets much smaller when omitting whitespace changes.

### CODE REVIEW

**Comment by geliu2016** [comment]
@wangke19 could you please take a view?

File: test/extended/apiserver/graceful_termination.go [hunk 1]
```diff
@@ -143,84 +145,109 @@ var _ = g.Describe("[sig-api-machinery][Feature:APIServer][Late]", func() {
...
+		logReaders := map[string]io.Reader{}
+		results := map[string]int{}
 
-				file, err := os.Open(path)
+		if *controlPlaneTopology == configv1.ExternalTopologyMode {
+			mgmtClusterOC := exutil.NewHypershiftManagementCLI("default").AsAdmin().WithoutNamespace()
+			pods, err := mgmtClusterOC.KubeClient().CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{LabelSelector: "hypershift.openshift.io/control-plane-component=kube-apiserver"})
+			o.Expect(err).To(o.BeNil())
+			for _, pod := range pods.Items {
+				fileName, err := mgmtClusterOC.Run("logs", "-n", pod.Namespace, pod.Name, "-c", "audit-logs").OutputToFile(pod.Name + "-audit.log")
```
Comment (p0lyn0mial) [inline test/extended/apiserver/graceful_termination.go:161]: I think it only gets the latest logs, during e2e tests APIs might be restarted multiple times. The intention of this test was to validate that during a termination we didn't receive late requests. Checking all available logs increases our chances to detect incorrect behavior.

I think we would have to list all available logs and retrieve them individually. 

I am okay with what we have now we could just improve it in the future.

**Comment by p0lyn0mial** [comment]
/assign @soltysh 

for approval

**Review by soltysh — APPROVED** [review]
/approve

File: test/extended/apiserver/graceful_termination.go (hunk 1 above)
Comment (alvaroaleman) [reply test/extended/apiserver/graceful_termination.go:161]: >  APIs might be restarted multiple times. 

Why would the api get restarted during e2e tests?

Comment (p0lyn0mial) [reply test/extended/apiserver/graceful_termination.go:161]: some tests do that, i.e. `etcd is able to vertically scale up and down with a single node`

**Comment by alvaroaleman** [comment]
Looks like the test works, it passed on https://github.com/openshift/origin/pull/27236
/retest

**Pull request merged**